### PR TITLE
[IA-2633] Script to cache docker images in GCE snapshots

### PR DESCRIPTION
**Pull request by kyuksel**
[JIRA ticket
](https://broadworkbench.atlassian.net/browse/IA-2633)

I need to merge this non-functional script addition PR first to be able to test the remaining changes end-to-end. Without it, some of the `Jenkinsfile` `git` commands fail because this file doesn't exist in `develop` branch yet.

### CODE REVIEW

**Review by rtitle — APPROVED** [review]

File: jenkins/gce-custom-images/prepare_gce_snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# [  "roles/compute.admin",
+#    "roles/iam.serviceAccountUser",
+#    "roles/storage.objectViewer",
+
+#
+# Constants and Global Vars
+#
+
+# The versions below don't matter; they are replaced by the Jenkins job
+terra_jupyter_base="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-base:0.0.19"
```
Comment (Qi77Qi) [inline jenkins/gce-custom-images/prepare_gce_snapshot.sh:18]: would be nice to switch these to some function that reads from automation tests's `reference.conf`...but doesn't have to be in this PR

Comment (rtitle) [reply jenkins/gce-custom-images/prepare_gce_snapshot.sh:18]: I think how it works is the Jenkins job replaces these version strings with `sed` when it builds the image

File: jenkins/gce-custom-images/prepare_gce_snapshot.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+set -e -x
+
+#
+# This script sets up a custom GCE snapshot for VMs provisioned by Leonardo.
```
Comment (Qi77Qi) [inline jenkins/gce-custom-images/prepare_gce_snapshot.sh:6]: can we rephrase this to `sets up a snapshot of boot disk` or something (there's nothing "custom" about the snapshot in that all snapshots are custom made anyway)....just don't want to confuse ppl with custom image.

File: jenkins/gce-custom-images/prepare_gce_snapshot.sh (hunk 1 above)
Comment (kyuksel) [reply jenkins/gce-custom-images/prepare_gce_snapshot.sh:18]: 👍🏼   will add a TODO

**Review by Qi77Qi — APPROVED** [review]

File: jenkins/gce-custom-images/prepare_gce_snapshot.sh (hunk 2 above)
Comment (kyuksel) [reply jenkins/gce-custom-images/prepare_gce_snapshot.sh:6]: Sure. I can change it in the main PR.
